x-pack/filebeat/input/httpjson/encoding_test.go

```diff
@@ -6,12 +6,32 @@ package httpjson
 
 import (
 	"encoding/json"
+	"io/ioutil"
 	"net/url"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
+func TestDecodeZip(t *testing.T) {
+	const expected = "[{\"a\":\"b\"},{\"a\":\"b\"},{\"c\":\"d\"},{\"a\":\"b\"},{\"c\":\"d\"}]"
+	b, err := ioutil.ReadFile("./testdata/test.zip")
```

Comment:
How about generating the zip content instead of committing a binary file? https://pkg.go.dev/archive/zip#example-Writer